packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/EmailInvoiceDetails.tsx

```diff
@@ -143,6 +143,18 @@ const EmailInvoiceDetailsBase = ({
     ]
   );
 
+  const isDisabled = useMemo(() => {
```

Comment:
@MadMed677, I'm already embarrassed to mention this every time, but such operations don't need to be memoized. Using `useMemo(...)` here only adds semantic load, and actually only slows down the code. I won't point out other places where you refactored and left the unreasonable use of `useMemo(...)` as it is, but it would be nice if you fixed that too.

Reply:
I do not agree with that point though. But to avoid any holy wars I will change it.
But I'll change my mind if you provide any proof in our code that this `useMemo` slows down the code and provides no benefits.

Commit: [7fda03b](https://github.com/team-monite/monite-sdk/pull/71/commits/7fda03b35d2791e88649cfac65cb1b2bbd8bc005)

Reply:
**From official documentation:**

[Should you add useMemo everywhere? ](https://react.dev/reference/react/useMemo#should-you-add-usememo-everywhere)

Optimizing with useMemo  is only valuable in a few cases:

- The calculation you’re putting in useMemo is noticeably slow, and its dependencies rarely change.
- You pass it as a prop to a component wrapped in [memo.](https://react.dev/reference/react/memo) You want to skip re-rendering if the value hasn’t changed. Memoization lets your component re-render only when dependencies aren’t the same.
- The value you’re passing is later used as a dependency of some Hook. For example, maybe another useMemo calculation value depends on it. Or maybe you are depending on this value from [useEffect.](https://react.dev/reference/react/useEffect)

There is no benefit to wrapping a calculation in useMemo in other cases. There is no significant harm to doing that either, so some teams choose to not think about individual cases, and memoize as much as possible. The downside of this approach is that code becomes less readable. Also, not all memoization is effective: a single value that’s “always new” is enough to break memoization for an entire component.

Reply:
@MadMed677, tanks, Solved.